com.unity.multiplayer.mlapi/Runtime/Interest/InterestManager.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections.Generic;
+using MLAPI.Connection;
+using UnityEngine;
+
+namespace MLAPI.Interest
+{
+    public class InterestManager
+    {
+        private readonly InterestNodeStatic m_DefaultInterestNode;
+
+        public InterestManager()
+        {
+            m_ChildNodes = new HashSet<InterestNode>();
+
+            // This is the node objects will be added to if no replication group is
+            //  specified, which means they always get replicated
+            m_DefaultInterestNode = ScriptableObject.CreateInstance<InterestNodeStatic>();
+            AddNode(m_DefaultInterestNode);
+        }
+
+        public void QueryFor(in NetworkClient client, HashSet<NetworkObject> results)
+        {
+            foreach (var c in m_ChildNodes)
+            {
+                c.QueryFor(client, results);
```

Comment:
maybe this PR is not the appropriate place to ask this but I'd like to see a sequence diagram or some sort of technical breakdown to understand how all these systems relate to each other and execute at runtime.

Reply:
Ok, sure, but in a nutshell, each interest node can have 0 or more child nodes, and it walks through them and calls the query function on each as it descends

Reply:
Admittedly, not sure I need the NetworkObject wrapper around it come to think of it though; likely you would trigger update on the Node itself and not need this convenience method.